cinema/views.py

```diff
@@ -322,56 +349,53 @@ def get_queryset(self):
-    model = Film
+def film_detail(request, pk):
+    film_list = Film.objects.select_related("imdb_rating", "mpaa_rating")
+    film = get_object_or_404(film_list, pk=pk)
+    film_info = info_by_films[pk]
+    context = {
+        'film': film, 'countries': sorted(film_info['country__name']),
+        'genres': sorted(film_info['genre__name']),
+        'languages': sorted(film_info['language__name']),
+        'distributors': sorted(film_info['distributor__name']),
+        'related_news': film_info['news'],
+        'film_persons': persons_by_films[pk]
+    }
+    comments = CommentToFilm.objects.filter(film=pk, is_active=True)
+    initial = {'film': film.pk}
+    if request.user.is_authenticated:
+        initial['author'] = request.user.username
+        form_class = UserCommentToFilmForm
+    else:
+        form_class = GuestCommentToFilmForm
+    form = form_class(initial=initial)
+
+    if request.method == 'POST':
+        c_form = form_class(request.POST)
+        if c_form.is_valid():
+            c_form.save()
+            messages.add_message(request, messages.SUCCESS, 'Comment added')
+        else:
+            form = c_form
+            messages.add_message(request, messages.WARNING, 'No Comment added')
 
-    def get_queryset(self):
-        return super().get_queryset().select_related(
-            "imdb_rating", "mpaa_rating"
-        )
+    context.update({'comments': comments, 'form': form})
 
-    def get_context_data(self, **kwargs):
-        context = super().get_context_data(**kwargs)
-        film_info = info_by_films[self.kwargs.get('pk')]
-        context['countries'] = sorted(film_info['country__name'])
-        context['genres'] = sorted(film_info['genre__name'])
-        context['languages'] = sorted(film_info['language__name'])
-        context['distributors'] = sorted(film_info['distributor__name'])
-        context['related_news'] = film_info['news']
-        context['film_persons'] = persons_by_films[self.kwargs.get('pk')]
-
-        # print(context)
-        # print(f'Атрибут "object" - это фильм "{self.object.title}", т.е. запись'
-        #       f' модели/таблицы БД "{self.object.__class__.__name__}"')
-        # print(self.get_template_names())
-        # print(self.template_name_suffix)
-        # print(self.get_context_object_name(self.object))
-        # print(self.kwargs)
-        # print(self.request)
-        # print(self.request.user.username)
-        # print(self.http_method_names)
-        # print(self.response_class)
-        # print(self.model)
-        # print(self.pk_url_kwarg)
-        # print(self.request.user.is_authenticated)
-        # print(self.request.get_host())
-        # print(
-        #     reverse_lazy('social:begin', args=('twitter',)),
-        #     reverse_lazy('social:complete', args=('twitter',))
-        # )
-
-        return context
+    return render(request, 'cinema/film_detail.html', context)
 
 
-class CinemaPersonDetailView(LoginRequiredMixin, DetailView):
```

Comment:
Why did you refuse django class bases views and switched on functions?

Reply:
Because, my controller must returns info about specific record of some model and, additionally, process comments form. Mr. Dronov recommends to use no mixed class controllers, at all! Better use simple function controllers. I agree with him absolutely.